test-suite/Math/NumberTheory/Moduli/PrimitiveRootTests.hs

```diff
@@ -90,6 +95,13 @@ isPrimitiveRootProperty4 (AnySign n) (Positive m)
     SomeMod n' -> not (isPrimitiveRoot n')
     InfMod{}   -> False
 
+isPrimitiveRootProperty5 :: Positive Natural -> Bool
+isPrimitiveRootProperty5 (Positive m)
```

Comment:
Let us use `Test.QuickCheck.Small` newtype here. `Positive` does not mean much for `Natural` and I am worried that if the `Arbitrary` instance of `Natural` gets changed, this test may take a really long time.

Reply:
I'm not entirely sure what your proposed fix here is, could you elaborate?

Reply:
My proposal was to write

```haskell
isPrimitiveRootProperty5 :: Small Natural -> Bool
isPrimitiveRootProperty5 (Smalll m) = ...
```

But now I realised that there is no `Small` newtype in `smallcheck` and my approach actually would not work. Nevermind.